problems/arraysWithLoops.js

```diff
@@ -25,39 +40,71 @@ function shoutWhileLoop() {}
  * @returns {Array} - ["I!", "am!", "a!", "happy!", "array!"]
  */
 
-function shoutForOfLoop() {}
+function shoutForOfLoop(Array) {
+  let arr = [];
+  for (const i of Array) {
+    arr.push(i + "!");
+  }
+  return arr;
+}
 
 /**
  * Takes in an array and returns the sum of all values
  * @param {number[]} nums
  * @returns {number} sum
  */
 
-function sumArray() {}
+function sumArray(nums) {
+  let sum = 0;
+  for (let i = 0; i < nums.length; i++) {
+    sum += nums[i];
+  }
+  return sum;
+}
 
 /**
  * Takes in an array and returns an array of all the odd valued elements
  * @param {number[]} nums
  * @returns {array} odds
  */
 
-function oddArray() {}
+function oddArray(nums) {
+  let arr = [];
+  for (let i = 0; i < nums.length; i++) {
+    if (nums[i] % 2 !== 0) {
+      arr.push(nums[i]);
+    }
+  }
+  return arr;
+}
 
 /**
  * Takes in an array and returns an array of all the even valued elements
  * @param {number[]} nums
  * @returns {array} evens
  */
 
-function evenArray() {}
+function evenArray(nums) {
+  let arr = [];
+  for (i = 0; i < nums.length; i++) {
+    if (nums[i] % 2 === 0) {
+      arr.push(nums[i]);
+    }
+  }
+  return arr;
+}
 
 /**
  * Takes in array and returns the smallest number in the array
  * @param {number[]} nums
  * @returns {number} smallest value
  */
 
-function findSmallest() {}
+function findSmallest(nums) {
+  for(let i = 0; i < nums.length; i++){
+    return Math.min(...nums)
```

Comment:
This is defeating the point of the problem. You could have just had this line. 
We're not using the loop at all. Instead you need to start with a variable that could be the smallest and compare each element in the array to that variable.